core/blockly.js

```diff
@@ -742,7 +742,7 @@ exports.zelos = zelos;
 // Blockly.Msg module - so make sure it is, but only if there is not
 // yet a Blockly global variable.
 if (!('Blockly' in globalThis)) {
```

Comment:
Is there a case where there was already a Blockly global variable *and* we need to be able to refer to it by `Blockly.Msg` instead of by importing it by name?

Reply:
I'm not sure I understand the question, so I will say a few things in the hopes that one of them answers it:

* At this point, barring an oversight on our part, none of the code in `core/`, `blocks/` or `generators/` ever assumes the existence of a `Blockly` global variable.
* Aside from (optionally named) `goog.require` statements in tests, etc., the only code in that should create a `Blockly` global is this hack that tries to ensure that `gloablThis.Blockly.Msg` (and only that) exists, and the call to `declareLegacyNamespace` at the top of this file that will create a global `Blockly` variable when this file is loaded in uncompiled mode or (with help from the UMD wrapper) if `blockly_compressed.js` is imported with a `<script>` tag into the browser.
* Most "outside" code should be importing Blockly via either an ES `import` statement or, if using Closure modules/compiler, a `goog.require` (e.g., Blockly Games and the advanced compilation test (ACT)).
  * I'm actually slightly unclear about importing Blockly using `import` works outside of node, as I wasn't aware that browsers have a way of loading UMD modules in response to ES `import` statements, but I would generally expect that no globals would be created as a result unless the import was being done as a named import at the top (script) level, in which case the _importer_ is probably going to name the import `Blockly`.
  * For BG and the ACT, AVANCED_OPTIMIZATIONS results in everything (including `Blockly`) being renamed, so there a bunch of random 1- and 2-character variable names that are meaningless without a sourcemap.
* In either case, unless the importer has gone out of their way to ensure that the `Blockly` global variable exists and points at the export object of this (`blockly.js` / `Blockly`) module, trying to load translations via `<script>` will fail.  This means that the following excerpt from `tests/compile/index.html`:
  ```
    <script src="main_compressed.js"></script>
    <script src="../../msg/js/en.js"></script>
  ```
  would throw when `en.js` tried to do `Blockly.Msg["ADD_COMMENT"] = "Add Comment"` because `Blockly.Msg` was undefined (until I fixed the issue in the first commit (8370647) of this PR).